src/pages/FacilityOrganization/components/FacilityOrganizationSelector.tsx

```diff
@@ -96,100 +93,89 @@ export default function FacilityOrganizationSelector(
   };
 
   const handleEdit = (level: number) => {
-    const newLevels = selectedLevels.slice(0, level);
-    setSelectedLevels(newLevels);
-    if (newLevels.length > 0) {
-      const lastOrg = newLevels[newLevels.length - 1];
-      setSelectedOrganization(lastOrg);
-      onChange(lastOrg.id);
-    } else {
-      setSelectedOrganization(null);
-    }
+    const orgList =
+      level === 0
+        ? getAllOrganizations?.results
+        : currentLevelOrganizations?.results;
+
+    const getDropdownLabel = () => {
+      if (level < selectedLevels.length) {
+        return selectedLevels[level].name;
+      }
+      return level === 0 ? t("select_department") : t("select_sub_department");
+    };
+
+    return (
+      <div className="group flex items-center gap-1.5">
+        {level > 0 && (
+          <CareIcon
+            icon="l-arrow-right"
+            className="h-3.5 w-3.5 text-gray-400 flex-shrink-0"
+          />
+        )}
+        <div className="flex-1 flex items-center gap-2">
+          <div className="flex-1">
+            <Autocomplete
+              value={selectedLevels[level]?.id}
+              options={getOrganizationOptions(orgList)}
+              onChange={(value) => handleLevelChange(value, level)}
+              placeholder={getDropdownLabel()}
+            />
+          </div>
+          {level > 0 && level < selectedLevels.length && (
+            <div
+              className="cursor-pointer p-1 hover:bg-gray-100 rounded-sm opacity-0 group-hover:opacity-100 transition-opacity"
+              onClick={() => {
+                const newLevels = selectedLevels.slice(0, level);
```

Comment:
This logic is what should be in a function called handleEdit 👍 

Separate out the concerns, code readability is important (check GovtOrganizationSelector for example for similar logic).